public/TCP/TCPClient.js

```diff
@@ -3,16 +3,20 @@ const ROVPort = 5000;
 const { encodeData, decodeData } = require('./coding');
 const { sendReceivedMessage } = require('./../IPC');
 
+// Variable for keeping track of connection status - used to handle ECONNREFUSED error
+let connected = false;
```

Comment:
Everything seems to work fine, but is `connected` ever set to `true`? I can't seem to find where it is, in any case.